packages/http/src/Session/VerifyCsrfMiddleware.php

```diff
@@ -67,9 +69,14 @@ private function ensureTokenMatches(Request $request): void
     {
         $tokenFromRequest = $request->get(
             key: Session::CSRF_TOKEN_KEY,
-            default: $request->headers->get(self::CSRF_HEADER_KEY),
         );
 
+        if (! $tokenFromRequest && $request->headers->has(self::CSRF_HEADER_KEY)) {
+            $tokenFromRequest = $this->encrypter->decrypt(
+                urldecode($request->headers->get(self::CSRF_HEADER_KEY)),
```

Comment:
I think it's fine to keep `urldecode` here, I don't see a reason why not if it helps reducing boilerplate on the front-end